app/controllers/members_controller.rb

```diff
@@ -24,12 +24,26 @@ def create
     end
   end
 
+  def verify_email
+    @member = Member.find_by_email_confirmation_token(params[:id])
+    if @member
+      @member.activate_account
+      MemberMailer.welcome_email(@member).deliver
+      flash[:success] = 'Welcome to the App! Your email has been confirmed.'
+      redirect_to action: :index
+    else
+      flash[:error] = 'Invalid token'
+      redirect_to action: :index
+    end
+  end
+
   def edit
     @member = Member.find(params[:id])
   end
 
   def update
     @member = Member.find(params[:id])
+    # params[:member][:email] = @member.email if @member.hide_email.eql? params[:member][:email]
```

Comment:
I'am really not sure how to write tests might take a day or two for me to figure it out.